src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java

```diff
@@ -0,0 +1,116 @@
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.OverseasEntityDueDiligenceDto;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.OverseasEntitySubmissionDto;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.DueDiligence;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.FilingForDate;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.Presenter;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.UpdateSubmission;
+
+/**
+ * This is a helper class for FilingsService when the filing is an update. The purpose of this class
+ * is to populate change submission values into UpdateSubmission.java with the ultimate gaol to
+ * produce a json output of change submissions. It uses OESubmissionDTO and UpdateDTO values to
+ * populate UpdateSubmission model.
+ */
+public class PopulateUpdateSubmission {
+  private OverseasEntitySubmissionDto overseasEntitySubmissionDto;
+  private UpdateSubmission updateSubmission;
+
+  public PopulateUpdateSubmission(
+      OverseasEntitySubmissionDto overseasEntitySubmissionDto, UpdateSubmission updateSubmission) {
+    this.overseasEntitySubmissionDto = overseasEntitySubmissionDto;
+    this.updateSubmission = updateSubmission;
+  }
+
+  /** method populates values into UpdateSubmission for JSON output */
+  public UpdateSubmission populate() {
+
+    updateSubmission.setUserSubmission(this.overseasEntitySubmissionDto);
+    populateDueDiligence(this.overseasEntitySubmissionDto, updateSubmission);
+    populatePresenter(this.overseasEntitySubmissionDto, this.updateSubmission);
+    updateSubmission.setBeneficialOwnerStatement(
+        this.overseasEntitySubmissionDto.getBeneficialOwnersStatement().getValue());
+
+    updateSubmission.setAnyBOsOrMOsAddedOrCeased(
+        this.overseasEntitySubmissionDto.getUpdate().isRegistrableBeneficialOwner());
+
+    return updateSubmission;
+  }
+
+  public void populateDueDiligence(
+      OverseasEntitySubmissionDto overseasEntitySubmissionDto, UpdateSubmission updateSubmission) {
+    DueDiligenceDto dueDiligenceDto = overseasEntitySubmissionDto.getDueDiligence();
+    OverseasEntityDueDiligenceDto overseasEntityDueDiligenceDto =
+        overseasEntitySubmissionDto.getOverseasEntityDueDiligence();
+    DueDiligence submissionDueDiligence = new DueDiligence();
+    submissionDueDiligence =
+        (overseasEntitySubmissionDto.getDueDiligence() != null)
+            ? populateByDueDiligenceDto(submissionDueDiligence, dueDiligenceDto)
+            : populateByOEDueDiligenceDto(submissionDueDiligence, overseasEntityDueDiligenceDto);
+    updateSubmission.setDueDiligence(submissionDueDiligence);
+
+  }
+
+  public DueDiligence populateByDueDiligenceDto(
+      DueDiligence dueDiligence, DueDiligenceDto dueDiligenceDto) {
+
+    dueDiligence.setDateChecked(dueDiligenceDto.getIdentityDate().toString());
```

Comment:
lines 59 to 64 and 74 to 79 could be split out to a common private method setDueDiligenceDetails()

Reply:
Venu - this comment is not resolved? There's quite a bit of duplication here.

Reply:
You would need to create a shared interface so not possible.